### PR TITLE
DB upgrade script

### PR DESCRIPTION
**Pull request by leonerd**
* Added a script to help AS authors upgrade databases from 0.3 to 1.0
 * Update documentation in `room-bridge-store.js` to more accurately reflect the data shape it creates

### CODE REVIEW

**Comment by Kegsay** [comment]
ok to test

File: scripts/upgrade-db-0.3-to-1.0.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+   --rooms   The path to rooms.db. Required.`
+);
+process.exit(0);
+}
+
+// *********************************************************
+// ** You will probably want to customise these functions **
+// *********************************************************
+
+function generateMatrixEntry(opts)
```
Comment (Kegsay) [inline scripts/upgrade-db-0.3-to-1.0.js]: Please doc what properties exist in `opts`.

If they are all the same then you can do it once in `:43`, but as of right now I got no clue.

File: scripts/upgrade-db-0.3-to-1.0.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+function generateRemoteEntry(opts)
+{
+    return null;
+}
+
+function generateLinkId(opts)
+{
+    return opts.matrix_id + " " + opts.remote_id;
```
Comment (Kegsay) [inline scripts/upgrade-db-0.3-to-1.0.js]: A one-liner explaining that the delimiter must not be a valid character in the `remote_id` format would be good.

File: scripts/upgrade-db-0.3-to-1.0.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+if (!opts.help && !opts.rooms) {
+    console.log("--rooms is required.");
+    opts.help = true;
+}
+
+if (opts.help) {
+    console.log(
+`Database Upgrade script (v0.3 => v1.0)
+--------------------------------------
+
```
Comment (Kegsay) [inline scripts/upgrade-db-0.3-to-1.0.js]: Where does this dump the result to?

File: scripts/upgrade-db-0.3-to-1.0.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            sparse: true
+        });
+    } catch (err) {
+        console.error(JSON.stringify(err));
+    }
+});
+
+Promise.coroutine(function*() {
+    try {
+        fs.mkdirSync("1.0-db");
```
Comment (Kegsay) [inline scripts/upgrade-db-0.3-to-1.0.js]: `rooms-1.0.db` may be a better name (`.db` file extension).

Comment (Kegsay) [inline scripts/upgrade-db-0.3-to-1.0.js]: Oh ignore me, this is the directory

**Comment by Kegsay** [comment]
LGTM aside from a tiny bit of extra documentation (given this is supposed to be read as a guide for upgrading).

File: scripts/upgrade-db-0.3-to-1.0.js (hunk 1 above)
Comment (leonerd) [inline scripts/upgrade-db-0.3-to-1.0.js]: Done 89ce7c8

File: scripts/upgrade-db-0.3-to-1.0.js (hunk 3 above)
Comment (leonerd) [inline scripts/upgrade-db-0.3-to-1.0.js]: Doced d819ec4

**Pull request merged**